server/src/test/java/io/crate/planner/operators/PushDownTest.java

```diff
@@ -109,14 +109,13 @@ public void testOrderByOnJoinWithMultipleRelationsPushedDown() {
                                 "inner join t1 as t3 on t3.a = t2.b " +
                                 "order by t2.b");
         assertThat(plan, isPlan(
-            "Eval[a, b, a]\n" +
-            "  └ NestedLoopJoin[INNER | (a = b)]\n" +
-            "    ├ NestedLoopJoin[INNER | (a = b)]\n" +
-            "    │  ├ OrderBy[b ASC]\n" +
-            "    │  │  └ Collect[doc.t2 | [b] | true]\n" +
-            "    │  └ Collect[doc.t1 | [a] | true]\n" +
-            "    └ Rename[a] AS t3\n" +
-            "      └ Collect[doc.t1 | [a] | true]"));
+            "OrderBy[b ASC]\n" +
```

Comment:
`OrderBy` is not pushed down, will have another look

Reply:
updated test to order by t1 to comply with the `MoveOrderBeneathNestedLoop` rule: order by relation must be topmost left. 

in master branch due to `HashSet` reordering t2 became topmost and was pushed down. When we use `LinkedHashSet` to ensure left_comes_first_in_set t1 is actual topmost and thus only order by t1 can be pushed down.